lldb/include/lldb/Host/PseudoTerminal.h

```diff
@@ -35,11 +35,14 @@ class PseudoTerminal {
 
   /// Destructor
   ///
-  /// The destructor will close the primary and secondary file descriptors if
-  /// they are valid and ownership has not been released using one of: @li
-  /// PseudoTerminal::ReleasePrimaryFileDescriptor() @li
-  /// PseudoTerminal::ReleaseSaveFileDescriptor()
-  ~PseudoTerminal();
+  /// The destructor will close the primary and secondary file
+  /// descriptor/HANDLEs if they are valid and ownership has not been released
+  /// using PseudoTerminal::Close().
+  virtual ~PseudoTerminal();
```

Comment:
I'm not sure if this pattern is the right thing. Either `PseudoTerminal` should become an ABC or we should use `#if defined` conditions. This hybrid mode is very subtle in the sense that I could add Unix specific behaviours and break windows without realising it.

Reply:
I agree. I think we should use `#if defined` instead, a lot of the methods in `PseudoTerminal` don't make sense on Windows. (GetPrimary and GetSecondary for instance).